tests/integration/security/jwt_test.go

```diff
@@ -464,83 +466,103 @@ func TestIngressRequestAuthentication(t *testing.T) {
+				},
+				{
+					Name:               "allow with sub-1 token",
+					Host:               "example.com",
+					Path:               "/",
+					Token:              jwt.TokenIssuer1,
+					ExpectResponseCode: 200,
+				},
+				{
+					Name:               "deny with sub-2 token",
+					Host:               "example.com",
+					Path:               "/",
+					Token:              jwt.TokenIssuer2,
+					ExpectResponseCode: 403,
+				},
+				{
+					Name:               "deny with expired token",
+					Host:               "example.com",
+					Path:               "/",
+					Token:              jwt.TokenExpired,
+					ExpectResponseCode: 401,
+				},
+				{
+					Name:               "allow with sub-1 token on any.com",
+					Host:               "any-request-principlal-ok.com",
+					Path:               "/",
+					Token:              jwt.TokenIssuer1,
+					ExpectResponseCode: 200,
+				},
+				{
+					Name:               "allow with sub-2 token on any.com",
+					Host:               "any-request-principlal-ok.com",
+					Path:               "/",
+					Token:              jwt.TokenIssuer2,
+					ExpectResponseCode: 200,
+				},
+				{
+					Name:               "deny without token on any.com",
+					Host:               "any-request-principlal-ok.com",
+					Path:               "/",
+					ExpectResponseCode: 403,
+				},
+				{
+					Name:               "deny with token on other host",
+					Host:               "other-host.com",
+					Path:               "/",
+					Token:              jwt.TokenIssuer1,
+					ExpectResponseCode: 403,
+				},
+				{
+					Name:               "allow healthz",
+					Host:               "example.com",
+					Path:               "/healthz",
+					ExpectResponseCode: 200,
+				},
+			}
+
+			for _, c := range ingTestCases {
+				echotest.New(t, apps.All).
+					SetupForDst(func(t framework.TestContext, dst echo.Instances) error {
+						policy := yml.MustApplyNamespace(t, tmpl.MustEvaluate(
+							file.AsStringOrFail(t, "testdata/requestauthn/ingress.yaml.tmpl"),
+							map[string]string{
+								"Namespace": ns.Name(),
+								"dst":       dst[0].Config().Service,
+							},
+						), ns.Name())
+						return t.Config().ApplyYAML(ns.Name(), policy)
+					}).
+					To(
+						echotest.SingleSimplePodServiceAndAllSpecial(),
+						echotest.Not(func(instances echo.Instances) echo.Instances { return instances.Match(echo.IsHeadless()) }),
+						echotest.Not(func(instances echo.Instances) echo.Instances { return instances.Match(echo.IsNaked()) }),
+						echotest.Not(func(instances echo.Instances) echo.Instances { return instances.Match(echo.IsExternal()) }),
+						echotest.Not(func(instances echo.Instances) echo.Instances { return instances.Match(util.IsMultiversion()) }),
+						func(instances echo.Instances) echo.Instances { return instances.Match(echo.Namespace(ns.Name())) },
+					).
+					RunFromClusters(func(t framework.TestContext, src cluster.Cluster, dest echo.Instances) {
+						ingr := ist.IngressFor(src)
```

Comment:
we should have `if ingr == nil { t.Skip() }` to be safe here. 

We could also bake that into the framework with a `RunFromIngress` that re-uses the `RunFromClusters`.

Reply:
Done.